activerecord/lib/active_record/database_configurations.rb

```diff
@@ -102,6 +113,7 @@ def env_with_configs(env = nil)
 
       def build_configs(configs)
         return configs.configurations if configs.is_a?(DatabaseConfigurations)
+        return configs if configs.is_a?(Array)
```

Comment:
If we're calling `configs_for =` we're replacing already created configurations and those will be an array.